crates/vsock/src/thread_backend.rs

```diff
@@ -78,7 +79,7 @@ pub(crate) struct VsockThreadBackend {
 impl VsockThreadBackend {
     /// New instance of VsockThreadBackend.
     pub fn new(
-        host_socket_path: String,
+        host_socket_path: impl AsRef<Path>,
```

Comment:
nit: This should be a `PathBuf` I think. Does not require generics and prevents an _actually_ unneeded copy. The user of this does have a PathBuf that is no longer needed and this is always needing to reconstruct a PathBuf from the ref.

Reply:
Switched everything to take a PathBuf